src/pages/images/actions/UploadInstanceBtn.tsx

```diff
@@ -0,0 +1,252 @@
+import { ChangeEvent, FC, useCallback, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  Input,
+  Modal,
+} from "@canonical/react-components";
+import { Form, useNavigate } from "react-router-dom";
+import { useFormik } from "formik";
+import * as Yup from "yup";
+import { instanceNameValidation, truncateInstanceName } from "util/instances";
+import { useProject } from "context/project";
+import { useQueryClient } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { uploadInstance } from "api/instances";
+import { UploadState } from "types/storage";
+import { useEventQueue } from "context/eventQueue";
+import { useToastNotification } from "context/toastNotificationProvider";
+import ProgressBar from "components/ProgressBar";
+import { humanFileSize } from "util/helpers";
+import usePortal from "react-useportal";
+import StoragePoolSelector from "pages/storage/StoragePoolSelector";
+import { AxiosError } from "axios";
+import { LxdSyncResponse } from "types/apiResponse";
+
+export interface UploadInstanceFormValues {
+  instanceFile: File | null;
+  name: string;
+  pool: string;
+}
+
+const UploadInstanceBtn: FC = () => {
+  const { openPortal, closePortal, isOpen, Portal } = usePortal();
+  const { project, isLoading: isProjectLoading } = useProject();
+  const [isUploading, setIsUploading] = useState(false);
+  const [uploadState, setUploadState] = useState<UploadState | null>(null);
+  const eventQueue = useEventQueue();
+  const toastNotify = useToastNotification();
+  const queryClient = useQueryClient();
+  const navigate = useNavigate();
+  const [uploadAbort, setUploadAbort] = useState<AbortController | null>(null);
+  const instanceNameAbort = useState<AbortController | null>(null);
+
+  const changeFile = async (e: ChangeEvent<HTMLInputElement>) => {
+    const { onChange } = formik.getFieldProps("instanceFile");
+    onChange(e);
+
+    if (e.currentTarget.files) {
+      const file = e.currentTarget.files[0];
+      const suffix = "-imported";
+      const instanceName = truncateInstanceName(
+        // remove file extension
+        file.name.split(".tar")[0],
```

Comment:
should we split on the first dot instead of assuming the file extension is .tar?
```suggestion
        file.name.split(".")[0],
```

Reply:
sure, I previously  thought `.` is a valid char in instance names